ext/CatalystHomotopyContinuationExtension/homotopy_continuation_extension.jl

```diff
@@ -0,0 +1,127 @@
+### Homotopy Continuation Based Steady State Finding ###
+
+"""
+    hc_steady_states(rs::ReactionSystem, ps; filter_negative=true, neg_thres=-1e-20, u0=typeof(ps)(), kwargs...)
+
+Uses homotopy continuation via HomotopyContinuation.jl to find the steady states of the ODE system corresponding to the provided reaction system.
+
+Arguments:
+- `rs::ReactionSystem`: The reaction system for which we want to find the steady states.
+- `ps`: The parameter values for which we want to find the steady states.
+- `filter_negative=true`: If set to true, solutions with any species concentration <neg_thres is removed from the output.
+- `neg_thres=-1e-20`: Determine the minimum values for which a species concentration is to be considred non-negative. Species conentrations ``> neg_thres`` but `< 0.0` are set to `0.0`.
+- `u0=nothing`: Initial conditions for which we want to find the steady states. For systems with conservation laws this are required to compute conserved quantities.
+- `kwargs...`: any additional arguments (like `show_progress= true`) are passed into HomotopyContinuation.jl's `solve` call. 
+
+Examples
+```@repl
+rs = @reaction_network begin
+    k1, Y --> 2X
+    k2, 2X --> X + Y
+    k3, X + Y --> Y
+    k4, X --> 0
+end
+ps = [:k3 => 1.0, :k2 => 2.0, :k4 => 1.5, :k1=>8.0]
+hc_sol = hc_steady_states(rs, ps)
+```
+gives
+```
+[0.5000000000000002, 2.0000000000000004]
+[0.0, 0.0]
+[4.499999999999999, 5.999999999999999]
+```
+
+Notes:
+```
+  """
+function Catalyst.hc_steady_states(rs::ReactionSystem, ps; filter_negative=true, neg_thres=-1e-20, u0=[], kwargs...)
+    ss_poly = steady_state_polynomial(rs, ps, u0)
+    sols = HC.real_solutions(HC.solve(ss_poly; kwargs...))
+    reorder_sols!(sols, ss_poly, rs)
+    return (filter_negative ? filter_negative_f(sols; neg_thres) : sols)
+end
+
+# For a given reaction system, paraemter values, and initial conditions, find the polynomial that HC solves to find steady states.
+function steady_state_polynomial(rs::ReactionSystem, ps, u0)
+    ns = convert(NonlinearSystem, rs; remove_conserved = true)
+    pre_varmap = map(pair -> pair, [symmap_to_varmap(rs,u0); symmap_to_varmap(rs,ps)]) 
```

Comment:
Do you have a test that this all works if the user passes a tuple for the parameter map using mixed types like integers and floats (even if the u0 map is a vector)?